pkg/ipamd/ipamd.go

```diff
@@ -863,14 +869,36 @@ func (c *IPAMContext) nodeIPPoolTooLow() bool {
 	total, used := c.dataStore.GetStats()
 
 	available := total - used
-	poolTooLow := available < c.maxIPsPerENI*c.warmENITarget || (c.warmENITarget == 0 && available == 0)
+	// instance max ENI is equal to or greater than 1, we want the smaller number of maxENI and WARM_ENI_TARGET.
+	poolTooLow := available < c.maxIPsPerENI * min(c.maxENI, c.warmENITarget) || (c.warmENITarget == 0 && available == 0)
```

Comment:
Please add testcases in the overview covering these scenarios -

Max eni is less than warm eni target
Max eni not being set
Max eni greater than warm eni target
very high max eni and warm eni target